ast/compile.go

```diff
@@ -883,6 +883,60 @@ func compileSchema(goSchema interface{}) (*gojsonschema.Schema, error) {
 	return schemasCompiled, nil
 }
 
+//Checks if a JSON schema contains a specific type
+func containsType(schema *gojsonschema.SubSchema, s string) bool {
+	return schema.Types.String() == s
+}
+
+//Merges JSON schemas
+func mergeSchemas(schemas []*gojsonschema.SubSchema) (*gojsonschema.SubSchema, error) {
+	var result = schemas[0]
+	if len(schemas) == 0 {
+		return result, nil
+	}
+
+	for i := 0; i < len(schemas); i++ {
+		if len(schemas[i].PropertiesChildren) > 0 {
+			if !schemas[i].Types.Contains("object") {
+				schemas[i].Types.Add("object")
+			}
+		} else if len(schemas[i].ItemsChildren) > 0 {
+			if !schemas[i].Types.Contains("array") {
+				schemas[i].Types.Add("array")
+			}
+		}
+	}
+
+	for i := 1; i < len(schemas); i++ {
+		if result.Types.Contains("object") && len(result.PropertiesChildren) > 0 && schemas[i].Types.Contains("object") && len(schemas[i].PropertiesChildren) > 0 {
+			for j := 0; j < len(schemas[i].PropertiesChildren); j++ {
+				result.PropertiesChildren = append(result.PropertiesChildren, schemas[i].PropertiesChildren[j])
+			}
+		} else if result.Types.Contains("array") && len(result.ItemsChildren) > 0 && schemas[i].Types.Contains("array") && len(schemas[i].ItemsChildren) > 0 {
+			for j := 0; j < len(schemas[i].ItemsChildren); j++ {
+				if result.ItemsChildren[j].Types.String() != schemas[i].ItemsChildren[j].Types.String() {
+					return nil, fmt.Errorf("unable to merge these schemas due to different types")
+				} else if !containsType(result.ItemsChildren[j], schemas[i].ItemsChildren[j].Types.String()) {
+					result.ItemsChildren = append(result.ItemsChildren, schemas[i].ItemsChildren[j])
+				}
+			}
+		} else if (result.Types.Contains("object") && schemas[i].Types.Contains("array")) || (result.Types.Contains("array") && schemas[i].Types.Contains("object")) {
```

Comment:
This if condition can be changed to 
```
if (result.Types.String() != schemas[i].Types.String()) {
  return nil, fmt.Errorf("unable to merge these schemas")
}
 ```
 
 because there are other types that can cause unmergeable status other than array and objects

Reply:
And this condition could be the first if condition check at line 911 in this series so that we dont go further with checking if types are different.